src/Core/Framework/DataAbstractionLayer/Search/RequestCriteriaBuilder.php

```diff
@@ -504,4 +513,59 @@ private function buildFieldName(EntityDefinition $definition, string $fieldName)
 
         return $fieldName;
     }
+
+    /**
+     * @return array<string, mixed>
+     */
+    private function decodeCriteriaParameter(string $encodedCriteria): array
+    {
+        try {
+            // Hard limit to avoid overloading
+            if (\strlen($encodedCriteria) > self::MAX_GET_CRITERIA_LENGTH) {
+                throw DataAbstractionLayerException::invalidCriteriaParameter('The _criteria parameter is too long');
+            }
+
+            // Decode base64url
+            $gzippedData = $this->base64urlDecode($encodedCriteria);
+            if ($gzippedData === false) {
+                throw DataAbstractionLayerException::invalidCriteriaParameter('Unable to decode base64 data');
+            }
+
+            // Decompress gzipped data
+            // Limit the decompressed size for additional safety from malicious input.
+            // Function throws a warning on failure, suppressing it as result is validated afterwards.
+            $jsonData = @gzdecode($gzippedData, self::MAX_GET_CRITERIA_UNCOMPRESSED_LENGTH);
+
+            if ($jsonData === false) {
+                throw DataAbstractionLayerException::invalidCriteriaParameter('Unable to decompress gzipped data');
+            }
+
+            // Decode JSON
+            $criteriaData = json_decode($jsonData, true, 512, \JSON_THROW_ON_ERROR);
+
+            if (!\is_array($criteriaData)) {
+                throw DataAbstractionLayerException::invalidCriteriaParameter('Criteria data must be an array');
+            }
+
+            return $criteriaData;
+        } catch (\JsonException $e) {
```

Comment:
instead of putting the try around everything, you should wrap only the lines, that are actually throwing those specific exceptions